hazelcast/src/main/java/com/hazelcast/instance/DefaultAddressPicker.java

```diff
@@ -198,6 +174,37 @@ private AddressDefinition pickInterfaceAddressDef() throws UnknownHostException,
         return interfaces;
     }
 
+    private Map<String, String> createAddressDomainMap(TcpIpConfig tcpIpConfig) {
+        Map<String, String> addressDomainMap;
+        if (!tcpIpConfig.isEnabled()) {
+            return Collections.emptyMap();
+        }
+        Collection<String> configuredMembers = TcpIpJoiner.getConfigurationMembers(config);
+        // LinkedHashMap is to guarantee order
+        addressDomainMap = createLinkedHashMap(configuredMembers.size());
+        for (String configuredMember : configuredMembers) {
+            String addressHolder = AddressUtil.getAddressHolder(configuredMember).getAddress();
+            if (AddressUtil.isIpAddress(addressHolder)) {
+                // there may be a domain registered for this address
+                if (!addressDomainMap.containsKey(addressHolder) && AddressUtil.matchAnyLocalInterface(addressHolder)) {
+                    addressDomainMap.put(addressHolder, null);
+                }
+            } else {
+                try {
+                    Collection<String> addresses = resolveDomainNames(addressHolder);
+                    for (String address : addresses) {
+                        if (AddressUtil.matchAnyLocalInterface(address)) {
```

Comment:
This doesn't seem efficient - it leads to too many calls of `matchersForAllLocalInterfaces` if there is more members (and they resolves to more IP adddresses).